cfbs/analyze.py

```diff
@@ -551,8 +558,51 @@ def analyze_policyset(
     if reference_version is None:
         reference_version = versions_data.version_counter.most_common_version()
 
-    # if not a single file in the analyzed policyset has an MPF-known checksum, and a specific `reference_version` was not given, `reference_version` will still be `None`
+    # if not a single file in the analyzed policyset has an MPF-known (checksum, filepath), and a specific `reference_version` was not given, `reference_version` will still be `None`
     if reference_version is None:
+        # try to detect whether the user provided a wrong policy set path
+        # gather all possible policy set paths, by detecting promises.cf or update.cf
+        possible_policyset_relpaths = []
+        mpfnorm_path = mpf_normalized_path(path, is_parentpath, masterfiles_dir)
+        for filepath in files_dict:
+            file_name = name(filepath)
+            if file_name in ("promises.cf", "update.cf"):
+                relpath = os.path.relpath(filepath, mpfnorm_path)
+                relpath = relpath.replace(os.sep, "/")
+
+                if "/" not in relpath:
+                    # `"."`, not `path`, as the list collects relative paths
+                    possible_policyset_relpaths.append(".")
+                if relpath.endswith("/update.cf"):
+                    possible_policyset_relpaths.append(relpath[:-10])
+                if relpath.endswith("/promises.cf"):
+                    possible_policyset_relpaths.append(relpath[:-12])
+
+        # for drive root, the path's parent is the path itself, so only check the parent path if this is not the case
+        if os.path.realpath(path) != os.path.realpath(os.path.join(path, "..")):
+            if os.path.exists(os.path.join(path, "..", "update.cf")) or os.path.exists(
+                os.path.join(path, "..", "promises.cf")
+            ):
+                possible_policyset_relpaths.append("..")
```

Comment:
`drive root` is the root of the filesystem, right?

This part is a bit cryptic, and could be more readable, I think, something like this;

```suggestion
        # for file system root, the path's parent is the path itself, so only check the parent path if this is not the case
        current = os.path.realpath(path)
        parent = os.path.realpath(os.path.join(path, ".."))
        at_root = (path == "/") # or current == parent
        if not at_root:
            if os.path.exists(os.path.join(parent, "update.cf")) or os.path.exists(
                os.path.join(parent, "promises.cf")
            ):
                possible_policyset_relpaths.append("..")
```

That might even make the comment unnecessary (which is a good thing).

Also, importing the functions you need from `os.path` could help reduce noise;

```
from os.path import exists, realpath, join
```